Marlin/src/HAL/STM32F1/fast_pwm.cpp

```diff
@@ -23,46 +23,46 @@
 
 #include "../../inc/MarlinConfigPre.h"
 
-#if NEEDS_HARDWARE_PWM
-
 #include <pwm.h>
 #include "HAL.h"
 #include "timers.h"
 
-void set_pwm_frequency(const pin_t pin, int f_desired) {
-  if (!PWM_PIN(pin)) return;                    // Don't proceed if no hardware timer
-
-  timer_dev *timer = PIN_MAP[pin].timer_device;
-  uint8_t channel = PIN_MAP[pin].timer_channel;
-
-  // Protect used timers
-  if (timer == get_timer_dev(TEMP_TIMER_NUM)) return;
-  if (timer == get_timer_dev(STEP_TIMER_NUM)) return;
-  #if PULSE_TIMER_NUM != STEP_TIMER_NUM
-    if (timer == get_timer_dev(PULSE_TIMER_NUM)) return;
-  #endif
-
-  if (!(timer->regs.bas->SR & TIMER_CR1_CEN))   // Ensure the timer is enabled
-    timer_init(timer);
-
-  timer_set_mode(timer, channel, TIMER_PWM);
-  uint16_t preload = 255;                       // Lock 255 PWM resolution for high frequencies
-  int32_t prescaler = (HAL_TIMER_RATE) / (preload + 1) / f_desired - 1;
-  if (prescaler > 65535) {                      // For low frequencies increase prescaler
-    prescaler = 65535;
-    preload = (HAL_TIMER_RATE) / (prescaler + 1) / f_desired - 1;
-  }
-  if (prescaler < 0) return;                    // Too high frequency
-  timer_set_reload(timer, preload);
-  timer_set_prescaler(timer, prescaler);
-}
-
 void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255*/, const bool invert/*=false*/) {
   timer_dev *timer = PIN_MAP[pin].timer_device;
```

Comment:
this code is not safe imo... `if (!PWM_PIN(pin)) return;` missing imo

Reply:
Agreed!

Reply:
lol 
LPC HAL:
#define PWM_PIN(P)            true // all pins are PWM capable